miri

```diff
@@ -57,6 +57,31 @@ TOOLCHAIN=$(cd "$MIRIDIR"; rustup show active-toolchain | head -n 1 | cut -d ' '
 COMMAND="$1"
 [ $# -gt 0 ] && shift
 
+# Run rustfmt on all tests and crates
+fmt() {
+    find "$MIRIDIR" -not \( -name target -prune \) -name '*.rs' \
+        | xargs rustfmt +$TOOLCHAIN --edition=2021 --config-path "$MIRIDIR/rustfmt.toml" "$@"
+}
+
+# Run clippy on all crates
+clippy() {
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/Cargo.toml --all-targets "$@"
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/ui_test/Cargo.toml --all-targets "$@"
+    $CARGO clippy $CARGO_EXTRA_FLAGS --manifest-path "$MIRIDIR"/cargo-miri/Cargo.toml "$@"
+}
+
+if [[ -f ".auto-everything" || -f ".auto-toolchain" ]] ; then
+    ./rustup-toolchain
+fi
+
+if [[ -f ".auto-everything" || -f ".auto-fmt" ]] ; then
+    fmt
+fi
+
+if [[ -f ".auto-everything" || -f ".auto-clippy" ]] ; then
+    clippy -- -D warnings
```

Comment:
This will not work, RUSTFLAGS and CARGO_EXTRA_FLAGS has not been computed yet. We deliberately compute this after the "early" commands since those commands should *not* affect the environment.

Reply:
of course I didn't test clippy... only fmt and toolchain -.-